File: src/rectangle_area_service/src/cpp/ras_client.cpp
#include <ros/ros.h>

#include "rectangle_area_service/RectangleAreaService.h"

bool get_input(int& a, int& b) {
  std::string a_str, b_str;

  std::cout << "width: ";
  std::cin >> a_str;

  try {
    a = std::stoi(a_str);
  }
  catch (...) {
    return false;
  }

  std::cout << "height: ";
  std::cin >> b_str;

  try {
    b = std::stoi(b_str);
  }
  catch (...) {
    return false;
  }

  return true;
}

int main(int argc, char* argv[]) {
  ros::init(argc, argv, "ras_client");

  ros::NodeHandle node;
  ros::ServiceClient client = node.serviceClient<rectangle_area_service::RectangleAreaService>("rectangle_area");

  while (ros::ok()) {
    rectangle_area_service::RectangleAreaService srv;
    int a, b;
    if (!get_input(a, b)) {
      break;
    }

    srv.request.width = a;
    srv.request.height = b;

    if (client.call(srv)) {
      std::cout << "area: " << (long int)srv.response.area << "\n";
    }
    else {
      ROS_ERROR("Failed to call the service");
    }
  }

  return 0;
}

File: src/rectangle_area_service/src/py/ras_client.py
#!/usr/bin/env python
import rospy
from rospy.exceptions import ROSInterruptException
from rectangle_area_service.srv import RectangleAreaService, RectangleAreaServiceRequest, RectangleAreaServiceResponse

def send_request(a, b):
	try:
		client = rospy.ServiceProxy("rectangle_area", RectangleAreaService)
		response = client(a, b)
		return response.area

	except rospy.ServiceException(e):
		print(f'Server error: {e}')

def node():
	rospy.init_node("ras_client", anonymous=False)

	rospy.wait_for_service("rectangle_area")

	while not rospy.is_shutdown():
		try:
			a = int(input("width: "))
			b = int(input("height: "))
		except:
			break

		sum = send_request(a, b)
		print(f'area: {sum}')


# Start the node if the file is executed
if __name__ == "__main__":
	try:
		node()
	except ROSInterruptException:
		pass
